### PR TITLE
Design navbar

### PR DESCRIPTION
**Pull request by ilarne**

### CODE REVIEW

**Review by motri — APPROVED** [review]
All tests pass! Looking good!

**Review by immafirestarter-zz — APPROVED** [review]
# All tests passing!!

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -1,27 +1,41 @@
...
-      <%= button_to 'Sign out', sign_out_path, method: :delete %>
-    <% else %>
-      <%= link_to 'Sign in', sign_in_path %>
-    <% end %>
+<body>
+  <nav>
+    <a href="/">
+      <div id="page-logo">
+      </div></a>
+      <% if signed_in? %>
```
Comment (neoeno) [inline app/views/layouts/application.html.erb]: Indentation is a bit off in this file — think this `if` and its `else` need their contents indenting another step.

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -1,27 +1,41 @@
...
-      Signed in as: <%= current_user.email %>
-      <%= button_to 'Sign out', sign_out_path, method: :delete %>
-    <% else %>
-      <%= link_to 'Sign in', sign_in_path %>
-    <% end %>
+<body>
+  <nav>
+    <a href="/">
+      <div id="page-logo">
+      </div></a>
```
Comment (neoeno) [inline app/views/layouts/application.html.erb]: Indentation a little funny here.

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -1,27 +1,41 @@
...
+      <% if signed_in? %>
+      <div class="dropdown">
+        <li class="dropbtn"><i class="fa fa-chevron-circle-down" style="font-size:20px"></i></li>
```
Comment (neoeno) [inline app/views/layouts/application.html.erb]: I'd put this `style=` attribute inside a class instead, e.g.

```html
<li class="dropbtn"><i class="fa fa-chevron-circle-down large-icon"></i></li>
```

```css
.large-icon {
  font-size: 20px;
}
```

It's good to keep all your styling in the CSS — so you always know where it's coming from! And this has the added benefit of being reusable elsewhere.

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -1,27 +1,41 @@
...
+        </div>
+      </div>
+      <li><a href="/" class="right-link" >Home</a></li>
+      <li><span class="right-link"> Signed in as: <%= current_user.email %></span></li>
 
-    <div id="flash">
-      <% flash.each do |key, value| %>
-        <div class="flash <%= key %>"><%= value %></div>
+      <% else %>
+      <li>  <%= link_to 'Sign in', sign_in_path, { :class=>"right-link" } %></li>
```
Comment (neoeno) [inline app/views/layouts/application.html.erb]: So there are a few rules around `li` elements you should be aware of.

* `<li>` elements must be directly inside a `<ul>` or an `<ol>` element.
* `<ul>` and `<ol>` elements *direct children* can only be `<li>` elements.

So these are valid:

```html
<ul>
  <li>Hello!</li>
  <li>Hi!</li>
</ul>

<ul>
  <li><div>Woo!</div></li>
</ul>
```

But these are not:

```html
<div>
  <li>Noo!</li>
</div>

<ul>
  <div>
    <li>Hey no!</li>
    <li>Oops!</li>
  </div>
</ul>
```

The logic being — `ul` is an Unordered List and lists can only contain `li` (List Items). Other arrangements 'don't make sense' according to the spec.

Take a look at this stackoverflow thread for some discussion on this: https://stackoverflow.com/questions/4230983/can-you-have-lis-without-them-being-under-a-ul-or-ol . You'll probably need to amend some of the CSS to make it still look alright after the changes!

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -1,27 +1,41 @@
...
       <% end %>
     </div>
-
-    <%= yield %>
-  </body>
+  </nav>
+  <div id="flash">
+    <% flash.each do |key, value| %>
+    <div class="flash <%= key %>"><%= value %></div>
+    <% end %>
```
Comment (neoeno) [inline app/views/layouts/application.html.erb]: Some more indentation

**Review by neoeno — CHANGES_REQUESTED** [review]
Great work here! Really glad to see it coming through and the CSS looks pretty good and even Railsy in places! Unfortunately I've ruined your day with a few comments about indentation and HTML validity. Sorry!

File: app/views/layouts/application.html.erb (hunk 3 above)
Comment (ilarne) [reply app/views/layouts/application.html.erb]: Hi Kay! We actually tried this first and it didn't work that time, though we didn't think to add an extra class just for styling icons. I've just changed it now and it still wasn't happy to co-operate unless we appended '!important' to the end to override any other styling....though we aren't sure where the other styling is coming from! Perhaps it's part of the external Font Awesome formatting (where we got the icon from)?

**Comment by ilarne** [comment]
@neoeno hey Kay, we've updated this with your suggestions! Let us know if anything else needs changing. Thanks a lot!

**Review by neoeno — APPROVED** [review]
Looks good! Let's get this merged!

If you're interested in front-end life then come find me sometime and we'll talk about how to make this with the more idiomatic `<li>` tags etc.

**Pull request merged**